### PR TITLE
correctly log the state of ssl_enabled during run

### LINKED ISSUE #169: SSL enabled reports ssl_enabled => false in the logs
This is my config:

```
input {
    http {
        port => 8080
        ssl_enabled => true
        ssl_key => "/etc/logstash/certs/logstash/logstash.pkcs8.key"
        ssl_certificate => "/etc/logstash/certs/logstash/logstash.crt"
        ssl_certificate_authorities => ["/etc/logstash/certs/ca/ca.crt"]
        ssl_client_authentication => required
    }
}

```
This is the start-up log:

```
Sep 26 16:53:01 file-2 logstash[7717]: [2023-09-26T16:53:01,099][INFO ][logstash.javapipeline    ][main] Pipeline started {"pipeline.id"=>"main"}
Sep 26 16:53:01 file-2 logstash[7717]: [2023-09-26T16:53:01,103][INFO ][logstash.inputs.http     ][main][4a7501c4d01aacf96cafea534fcafd2c4bdd84a4a49fe92b28275bac53e3662f] Starting http input listener {:address=>"0.0.0.0:8080", :ssl=>"false"}
Sep 26 16:53:01 file-2 logstash[7717]: [2023-09-26T16:53:01,153][INFO ][logstash.agent           ] Pipelines running {:count=>1, :running_pipelines=>[:main], :non_running_pipelines=>[]}
```

### PR DESCRIPTION
**Pull request by jsvd**
fixes #169

### CODE REVIEW

File: spec/inputs/http_spec.rb [hunk 1]
```diff
@@ -544,6 +544,24 @@ def setup_server_client(url = self.url)
...
+        allow(http_server).to receive(:close)
+        allow(http_server).to receive(:run)
+        http_server
+      }
+      before(:each) do
+        allow(subject).to receive(:create_http_server).and_return(http_server)
+        subject.register
+      end
+      it "should show ssl is disabled" do
+        #[2024-10-18T10:09:33,199][INFO ][logstash.inputs.http     ][main][0d48] Starting http input listener {:address=>"0.0.0.0:3333", :ssl_enabled=>true}
```
Comment (kaisecheng) [inline spec/inputs/http_spec.rb:559]: do you want to keep this line? If so, it should be :ssl_enabled => false

File: spec/inputs/http_spec.rb [hunk 2]
```diff
@@ -585,6 +603,24 @@ def setup_server_client(url = self.url)
...
+          allow(http_server).to receive(:close)
+          allow(http_server).to receive(:run)
+          http_server
+        }
+        before(:each) do
+          allow(subject).to receive(:create_http_server).and_return(http_server)
+          subject.register
+        end
+        it "should show ssl is enabled" do
+          #[2024-10-18T10:09:33,199][INFO ][logstash.inputs.http     ][main][0d48] Starting http input listener {:address=>"0.0.0.0:3333", :ssl_enabled=>false}
```
Comment (kaisecheng) [inline spec/inputs/http_spec.rb:618]: :ssl_enabled => true

File: spec/inputs/http_spec.rb (hunk 1 above)
Comment (jsvd) [reply spec/inputs/http_spec.rb]: I prefer to keep it, the incongruence has been fixed :+1:

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## 3.9.1
+ - Correctly log the state of `ssl_enabled` during `run`
```
Comment (jsvd) [inline CHANGELOG.md:2]:
```suggestion
 - Correctly log the state of `ssl_enabled` during `run` [#180](https://github.com/logstash-plugins/logstash-input-http/pull/180)
```

**Review by kaisecheng — APPROVED** [review]
LGTM

**Pull request merged**